File: src/main/java/com/wlyscraft/slimefunitemexpansion/items/weapons/LightningBow.java
package com.wlyscraft.slimefunitemexpansion.items.weapons;

import io.github.thebusybiscuit.slimefun4.core.handlers.BowShootHandler;
import io.github.thebusybiscuit.slimefun4.implementation.items.weapons.SlimefunBow;
import me.mrCookieSlime.Slimefun.Objects.Category;
import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
import org.bukkit.inventory.ItemStack;

import javax.annotation.Nonnull;

public class LightningBow extends SlimefunBow {

    public LightningBow(Category category, SlimefunItemStack item, ItemStack[] recipe) {
        super(category, item, recipe);
    }

    @Nonnull
    @Override
    public BowShootHandler onShoot() {
        return (e, target) -> {
            if (!e.isCancelled()) target.getWorld().strikeLightning(target.getLocation());
        };
    }
}

File: src/main/kotlin/io/github/itsflicker/slimefunitemexpansion/items/weapons/LightningBow.kt
package io.github.itsflicker.slimefunitemexpansion.items.weapons

import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup
import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack
import io.github.thebusybiscuit.slimefun4.core.handlers.BowShootHandler
import io.github.thebusybiscuit.slimefun4.implementation.items.weapons.SlimefunBow
import org.bukkit.inventory.ItemStack

class LightningBow(itemGroup: ItemGroup, item: SlimefunItemStack, recipe: Array<ItemStack?>) :
    SlimefunBow(itemGroup, item, recipe) {

    override fun onShoot(): BowShootHandler {
        return BowShootHandler { e, target ->
            if (!e.isCancelled) target.world.strikeLightning(target.location)
        }
    }
}